src/Models/Model.php

```diff
@@ -326,10 +327,48 @@ public static function propertyKeys() : array {
 		return array_keys( ( new static() )->properties );
 	}
 
+	/**
+	 * Will infer the type based on the property definition and
+	 * cast the type on the value passed in.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key   The property name.
+	 * @param mixed  $value The value to be cast.
+	 *
+	 * @return array|bool|int|string The $value after being cast to it's type.
+	 */
+	protected function castAttribute( string $key, $value ) {
+		// Handle nullable fields as expected.
+		if ( is_null( $value ) ) {
+			return null;
+		}
+
+		$type = $this->getPropertyType( $key );
+
+		switch ( $type ) {
```

Comment:
Maybe changing it with `settype( $value, $type );` for multiple cases will be nicer, so you don't need to read all cases really as they are the same.